perfkitbenchmarker/log_util.py

```diff
@@ -0,0 +1,201 @@
+# Copyright 2014 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Utilities related to loggers and logging.
+"""
+
+import copy
+import logging
+import threading
+
+DEBUG = 'debug'
+INFO = 'info'
+LOG_LEVELS = {
+    DEBUG: logging.DEBUG,
+    INFO: logging.INFO
+}
+
+
+thread_local = threading.local()
+
+
+class ThreadLogInfo(object):
+  """Per-thread log message prefix label configuration."""
+  def __init__(self, thread_log_info=None):
+    """Constructs a ThreadLogInfo by copying a previous ThreadLogInfo.
+
+    Args:
+      thread_log_info: A ThreadLogInfo for an existing thread whose state will
+        be copied to initialize a ThreadLogInfo for a new thread.
+    """
+    if thread_log_info:
+      self._label_list = copy.copy(thread_log_info._label_list)
```

Comment:
For a shallow list copy, prefer `thread_log_info._label_list[:]` or `list(thread_log_info._label_list)`.